contracts/BC_fusion/StakePool.sol

```diff
@@ -0,0 +1,248 @@
+        (bool success,) = _delegator.call{value: _totalBnbAmount}("");
+        require(success, "CLAIM_FAILED");
+
+        emit UnbondClaimed(_delegator, _totalShares, _totalBnbAmount);
+        return _totalBnbAmount;
+    }
+
+    function distributeReward() external payable onlyStakeHub {
+        uint256 _bnbAmount = msg.value;
+        _totalReceivedReward += _bnbAmount;
+        _totalPooledBNB += _bnbAmount;
+        emit RewardReceived(_bnbAmount);
+    }
+
+    function slash(uint256 _slashBnbAmount) external onlyStakeHub returns (uint256) {
+        uint256 selfDelegation = _sharesOf(validator);
+        uint256 _slashShares = getSharesByPooledBNB(_slashBnbAmount);
+
+        uint256 _remainingSlashBnbAmount = _slashBnbAmount;
+        uint256 _remainingSlashShares = _slashShares;
+        if (_slashShares <= selfDelegation) {
+            _totalPooledBNB -= _slashBnbAmount;
+            _burn(validator, _slashShares);
+            _remainingSlashBnbAmount = 0;
+        } else {
+            uint256 selfDelegationTokens = getPooledBNBByShares(selfDelegation);
+            _totalPooledBNB -= selfDelegationTokens;
+            _burn(validator, selfDelegation);
+
+            _remainingSlashBnbAmount -= selfDelegationTokens;
+            _remainingSlashShares -= selfDelegation;
+
+            uint256 _unbondingShares = _lockedShares[validator];
+            while (_remainingSlashBnbAmount > 0 && _unbondingShares > 0) {
+                bytes32 hash = _unbondRequestsQueue[validator].front();
+                UnbondRequest memory request = _unbondRequests[hash];
+
+                if (request.bnbAmount > _remainingSlashBnbAmount) {
+                    // slash the request
+                    _unbondRequests[hash].sharesAmount -= _remainingSlashShares;
+                    _unbondRequests[hash].bnbAmount -= _remainingSlashBnbAmount;
+                    _unbondingShares -= _remainingSlashShares;
+                    _remainingSlashBnbAmount = 0;
+                    break;
+                } else {
+                    // slash the request and remove from the queue
+                    _unbondingShares -= request.sharesAmount;
+                    _remainingSlashBnbAmount -= request.bnbAmount;
+                    _remainingSlashShares -= request.sharesAmount;
+                    _unbondRequestsQueue[validator].popFront();
+                    delete _unbondRequests[hash];
+                }
+            }
+
+            _lockedShares[validator] = _unbondingShares;
+        }
+
+        uint256 _realSlashBnbAmount = _slashBnbAmount - _remainingSlashBnbAmount;
+        (bool success,) = SYSTEM_REWARD_ADDR.call{value: _realSlashBnbAmount}("");
+        require(success, "TRANSFER_FAILED");
+        return _realSlashBnbAmount;
+    }
+
+    /*----------------- view functions -----------------*/
+    function totalReceivedReward() external view returns (uint256) {
+        return _getTotalReceivedReward();
```

Comment:
why not return `_totalReceivedReward` directly, too many useless wrapper for some function.

Reply:
Done